src/realm/replication.hpp

```diff
@@ -474,18 +474,25 @@ inline void Replication::select_table(const Table* table)
 
 inline bool Replication::select_collection(const CollectionBase& coll)
 {
+    ConstTableRef table = coll.get_table();
+    ColKey col_key = coll.get_col_key();
+    ObjKey obj_key = coll.get_owner_key();
+    auto path = coll.get_stable_path();
```

Comment:
Reading the values here and passing them to do_select_collection() rather than passing the collection in is a bunch of extra instructions for no benefit and the structure of this means that the compiler can't inline it away.

The idea behind this division is that select_collection() gets inlined into each of the callers so that no function call is needed in the common case of the collection already being selected, but the less common case is outlined rather than being repeated for each called. This is a fairly dubious optimization that we haven't actually measured in the last decade, but it theoretically could be useful.

Reply:
Good observation. I cleaned it up a bit, hopefully this makes it easier for the compiler to make optimizations.